analysers/analyser_osmosis_highway_vs_building.py

```diff
@@ -26,10 +26,14 @@
 
 sql00 = """
 CREATE TEMP TABLE {0}highway AS
+WITH ways AS (
 SELECT
     id,
     linestring,
     ST_Transform(linestring, {1}) AS linestring_proj,
+    ST_Length(ST_Transform(linestring, {1})) AS length,
+    ceil(ST_Length(ST_Transform(linestring, {1})) / 500)::integer AS split_n,
+    ST_Length(ST_Transform(linestring, {1})) / ceil(ST_Length(ST_Transform(linestring, {1})) / 500)::integer AS split_length,
```

Comment:
`length` and `split_length` are only used in the combination `split_length / length`. However, given how they are defined, `split_length / length` = `1 / split_n` (i.e. the fraction of the linestring length).

Maybe it is clearer to omit `length` and `split_length` and use
`1/ceil(ST_Length(ST_Transform(linestring, {1})) / 500) AS split_length_fraction,` instead
(or `1 / split_n` in the code below instead of `split_length / length`)

Reply:
Right, fixed.